base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -3966,6 +3981,168 @@ where
+
+        // Store the senders for the new transaction
+        self.pending_transaction_reply_senders
+            .insert(new_tx_id, tx_reply_sender);
+        self.send_transaction_cancellation_senders
+            .insert(new_tx_id, cancellation_sender);
+
+        // Create new transaction protocol with updated fee
+        let protocol = TransactionSendProtocol::new(
+            new_tx_id,
+            self.resources.clone(),
+            tx_reply_receiver,
+            cancellation_receiver,
+            destination.clone(),
+            original_amount,
+            fee_per_gram, // This is the new, higher fee
+            payment_id.clone(),
+            TransactionMetadata::default(),
+            None, // No reply channel for internal call
+            TransactionSendProtocolStage::Initial,
+            None,
+        );
+
+        println!("Executing transaction send protocol");
+        // Launch the new transaction protocol
+        let join_handle = tokio::spawn(protocol.execute(Some(UtxoSelectionCriteria::must_include(original_inputs))));
+        send_transaction_join_handles.push(join_handle);
+
+        info!(
+            target: LOG_TARGET,
+            "Replace-by-fee: Created new transaction {} to replace {} with fee_per_gram: {}",
+            new_tx_id,
+            tx_id,
+            fee_per_gram
+        );
+
+        Ok(new_tx_id)
+    }
+
+    async fn user_pay_for_fee(
+        &mut self,
+        tx_id: TxId,
+        destination: TariAddress,
+        amount: MicroMinotari,
+        send_transaction_join_handles: &mut FuturesUnordered<
+            JoinHandle<Result<TransactionSendResult, TransactionServiceProtocolError<TxId>>>,
+        >,
+    ) -> Result<TxId, TransactionServiceError> {
+        let original_transaction = self
+            .resources
+            .db
+            .get_pending_outbound_transaction(tx_id)
+            .map_err(TransactionServiceError::TransactionStorageError)?;
+
+        // Check if transaction is already confirmed
+        if original_transaction.status.is_confirmed() {
+            return Err(TransactionServiceError::TransactionAlreadyCompleted(tx_id.to_string()));
+        }
+
+        let new_tx_id = TxId::new_random();
+        let (tx_reply_sender, tx_reply_receiver) = mpsc::channel(100);
+        let (cancellation_sender, cancellation_receiver) = oneshot::channel();
+
+        // Store the senders for the new transaction
+        self.pending_transaction_reply_senders
+            .insert(new_tx_id, tx_reply_sender);
+        self.send_transaction_cancellation_senders
+            .insert(new_tx_id, cancellation_sender);
+
+        // Create new transaction protocol with new outputs
+        let protocol = TransactionSendProtocol::new(
+            new_tx_id,
+            self.resources.clone(),
+            tx_reply_receiver,
+            cancellation_receiver,
+            destination.clone(),
+            amount,
+            original_transaction.fee,
```

Comment:
The `fee_per_gram: MicroMinotari,` for the new transaction cannot be equal to the entire fee of the original transaction.
```suggestion
            some_other_valur_for_fee_per_gram,
```

Reply:
Fixed. Now this value is calculated based on original transaction to get it's total weight.